proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,501 @@
+requests, we can revisit what the in flight limit should be.
+
+<a name="rationale-require-issues"></a>
+### Q. Why do we require issues to be opened instead of just pull requests?
+
+Having a defined process for intake removes any ambiguity on how to initiate a
+contribution by ensuring that all potential contributions start out as a GitHub
+issue. We feel that having a more general discussion about the problem and
+solutions will help reduce the overall effort into code review and provide
+contributors with enough detail to make a sucessful contribution.
+
+This requirement also solidifies the position that all pull requests that are
+currently open are something we'd like to pull in. It removes any ambiguity for
+a contributor about the status of the request.
+
+Pull requests are often made for problems that only affect the contributor or a
+small portion of the user base. An issue provides a mechanism to gather
+quantitative feedback in the form of "upvotes" through GitHub reactions to
+estimate the impact of the issue on the community.
+ 
+We intend for issues to be a way to improve contributor confidence in both their
+contributions and the overall process. Draft pull requests can be used to
+demonstrate a potential implementation and get community feedback or interest,
+but they may not be reviewed by the maintainers. If a user does open a pull
+request without an issue, we will use the opportunity to educate them about our
+process, and when necessary assist them in opening an issue.
+
+### Q. Why would we re-prioritize contributions for review instead of taking them first in, first out?
+
+By default issues should be selected for review based on the stated criteria,
+and existing issues should not be pulled up above other existing issues.
+However, there are scenarios where a new issue may be added to the queue in an
+'insertion sort' fashion. One example considers if the issues currently
+prioritized for review are all large and complex. In this case, they risk
+blocking smaller and easier changes. We may pull up the smaller issues to keep
+the queue moving. In another case, a new feature may increase in urgency due to
+other API changes. Since the main responsibility of the AWS CLI is to facilitate
+access to the AWS API, we would re-prioritize a feature request to address that
+change.
+
+<a name="rationale-upvotes"></a>
+### Q. Why should there be a minimum number of upvotes or reactions for a feature request?
+
+Feature requests carry a risk of adding new technical debt and increased
+maintenance cost and should have a higher bar to be included.
+
+The request should have user impact beyond the individual who proposed it. This
+is difficult to measure objectively, but one proxy for impact can be the number
+of upvote reactions on the issue. This is intended to prevent narrowly scoped
+changes that only affect a single user. It also helps to limit the number of in
```

Comment:
@kyleknap I feel like the upvotes limiting in flight reviews is not a good rationale. It's more protective of a process that we have other ways to manage; we're already up front with needing to balance other work and limiting the queue size. Proposing to drop this from the rationale.

Reply:
Yeah that sounds good to me. I honestly missed that point in the last pass I made.